File: test-fixtures/cross-language/typescript/src/dup_a.ts
// Code-duplication: identical function body to dup_b.ts
export function duplicateFunctionA(input: string): string {
  let result = '';
  for (const ch of input) {
    if (/[a-zA-Z]/.test(ch)) {
      result += ch.toUpperCase();
    } else if (/[0-9]/.test(ch)) {
      result += ch;
    } else {
      result += '_';
    }
  }
  return result;
}

export function anotherUniqueA(x: number): number {
  return x * x + 2 * x + 1;
}

File: test-fixtures/cross-language/rust/src/dup_a.rs
// Code-duplication: identical function body to dup_b.rs
pub fn duplicate_function_a(input: &str) -> String {
    let mut result = String::new();
    for ch in input.chars() {
        if ch.is_alphabetic() {
            result.push(ch.to_ascii_uppercase());
        } else if ch.is_numeric() {
            result.push(ch);
        } else {
            result.push('_');
        }
    }
    result
}

pub fn another_unique_function_a(x: i32) -> i32 {
    x * x + 2 * x + 1
}
